### PR TITLE
Added template engine

### PR DESCRIPTION
**Pull request by notzippy**

### CODE REVIEW

File: manual/templates.md [hunk 1]
```diff
@@ -9,14 +9,59 @@ godoc:
...
+play like
+
+```
+template.engines=pongo2,ace,go
+```
+
+The three template engines `pongo2,ace,go` will be used when rendering
+templates. 
+
+**NOTE** Ace and Pongo2 are not available at this time, but will be added
```
Comment (brendensoares) [inline manual/templates.md]: Or are they?

File: _layouts/manual.html [hunk 2]
```diff
@@ -37,6 +37,9 @@
     title: Results & Response
     url: results.html
   -
+    title: Templates Engines
```
Comment (brendensoares) [inline _layouts/manual.html]: Template

File: _layouts/manual.html [hunk 3]
```diff
@@ -37,6 +37,9 @@
...
+    url: template-engine.html
```
Comment (brendensoares) [inline _layouts/manual.html]: Match the title wording?

File: manual/template-engine.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+---
+title: Template Engine
```
Comment (brendensoares) [inline manual/template-engine.md]: Match the section title

File: manual/template-engine.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+github:
+  labels:
+    - topic-template
+godoc:
+    - TemplateEngine
+---
+
+Revel now allows for using custom template engines, the process for building your own is as follows
+
+###The TemplateEngine
```
Comment (brendensoares) [inline manual/template-engine.md]: Template Engine

File: manual/template-engine.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+A template engine must implement the [`TemplateEngine`](https://godoc.org/github.com/revel/revel#TemplateEngine)
+interface. To register a new engine in revel :
+ - Define it as a module to be loaded in the app.conf.
```
Comment (brendensoares) [inline manual/template-engine.md]: `app.conf`

File: manual/template-engine.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    - topic-template
+godoc:
+    - TemplateEngine
+---
+
+Revel now allows for using custom template engines, the process for building your own is as follows
+
+###The TemplateEngine
+A template engine must implement the [`TemplateEngine`](https://godoc.org/github.com/revel/revel#TemplateEngine)
+interface. To register a new engine in revel :
```
Comment (brendensoares) [inline manual/template-engine.md]: Revel

File: manual/template-engine.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+ - Define it as a module to be loaded in the app.conf.
+ - In an init function, which registers the engine in revel by calling:
```
Comment (brendensoares) [inline manual/template-engine.md]: `init()`...Revel

File: manual/template-engine.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+- [revel.IsEngineFor(engine TemplateEngine, templateView *TemplateView) bool](https://godoc.org/github.com/revel/revel#TemplateEngine)
+ Called for every view found, revel creates a [`Trevel.TemplateView`](https://godoc.org/github.com/revel/revel#TemplateView)
+ instance and calls every engine to see if they are the one to be used to parse the view
+
+- [ParseAndAdd(basePath *TemplateView) error](https://godoc.org/github.com/revel/revel#TemplateEngine)
+  Called on application startup, or when templates have been changed. 
+  
+- [Name() string](https://godoc.org/github.com/revel/revel#TemplateEngine)
+  Called to fetch the name of the TemplateEngine.
```
Comment (brendensoares) [inline manual/template-engine.md]: TemplateEngine?

File: manual/template-engine.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+ instance and calls every engine to see if they are the one to be used to parse the view
+
+- [ParseAndAdd(basePath *TemplateView) error](https://godoc.org/github.com/revel/revel#TemplateEngine)
+  Called on application startup, or when templates have been changed. 
+  
+- [Name() string](https://godoc.org/github.com/revel/revel#TemplateEngine)
+  Called to fetch the name of the TemplateEngine. 
+  
+- [Lookup(templateName string) revel.Template](https://godoc.org/github.com/revel/revel#TemplateEngine)
+  Called to fetch the Template, see below for what the template interface implements.
```
Comment (brendensoares) [inline manual/template-engine.md]: `Template`?

File: manual/templates-go.md [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+layout: manual
+github:
+  labels:
+    - topic-template
+godoc:
+    - GoEngine
+---
+
+## Go Template Engine
+The GO template engine is the default one used by Revel. It requires no application
```
Comment (brendensoares) [inline manual/templates-go.md]: Go or GO?

File: manual/templates-go.md [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+{% endraw %}{% endcapture %}
+{% highlight htmldjango %}{{ex}}{% endhighlight %}
+
+<div class="alert alert-info">Note: Paths are relative to <code>app/views</code></div>
+
+
+<a name="functions"></a>
+
+## GO Template Functions
+These builtin functions are only applicable to the GO template engine, other template
```
Comment (brendensoares) [inline manual/templates-go.md]: built-in

File: modules/index.md [hunk 13]
```diff
@@ -26,6 +26,15 @@ In order to add a module to your app, add a line to [`conf/app.conf`](appconf.ht
 module.mymodulename = go/import/path/to/module
 {% endhighlight %}
 
+You can add an ordering method by insert a sortable string in between the
+word module and the name of the module like the following. This allows you to specify 
+which order the views should be loaded, so your application will always be the first,
+then the order of the modules and finally the default views in revel.
```
Comment (brendensoares) [inline modules/index.md]: Revel

File: manual/template-engine.md (hunk 4 above)
Comment (notzippy) [inline manual/template-engine.md]: Changed title to **Template Engine** as well

**Comment by pedromorgan** [comment]
looks fine  2 me ..

**Pull request merged**